middle_end/flambda2.0/terms/flambda_primitive.ml

```diff
@@ -993,26 +1036,30 @@ let args_kind_of_ternary_primitive p =
   | Bytes_or_bigstring_set (Bigstring, Sixty_four) ->
     bigstring_kind, array_like_thing_index_kind,
       K.naked_int64
+  | Bigarray_set (_, kind, _) ->
+    let new_value = element_kind_of_bigarray_kind kind in
+    bigarray_kind, array_like_thing_index_kind, new_value
 
 let result_kind_of_ternary_primitive p : result_kind =
   match p with
   | Block_set _
-  | Bytes_or_bigstring_set _ -> Unit
+  | Bytes_or_bigstring_set _
+  | Bigarray_set _ -> Unit
 
 let effects_and_coeffects_of_ternary_primitive p =
   match p with
   | Block_set _
-  | Bytes_or_bigstring_set _ -> writing_to_an_array_like_thing
+  | Bytes_or_bigstring_set _
+  | Bigarray_set _ -> writing_to_an_array_like_thing
```

Comment:
Here and anywhere else this occurs, I would make a new function `writing_to_a_bigarray` or similar.  `Bigarray`s are complicated, and conflating the cases might lead us to miss something, even if the behaviour currently should be the same.

Reply:
done